### PR TITLE
[libcxxabi] Always link against libzircon for Fuchsia targets

### PR DESCRIPTION
**Pull request by ilovepi**
When using LTO, the deplibs mechanism is insufficient to ensure that libzircon
is always brought into the link prior to LTO code generation. The general
problem is discussed in depth in
https://github.com/llvm/llvm-project/issues/56070

To work around this, we can just provide libzircon as a link input.

### CODE REVIEW

**Review by petrhosek — APPROVED** [review]

File: libcxxabi/src/CMakeLists.txt [hunk 1]
```diff
@@ -75,6 +75,10 @@ if (NOT APPLE) # On Apple platforms, we always use -nostdlib++ so we don't need
   add_library_flags_if(LIBCXXABI_HAS_C_LIB c)
 endif()
 
+if (FUCHSIA)
+    add_link_flags("-Wl,--push-state,--as-needed,-lzircon,--pop-state")
```
Comment (petrhosek) [inline libcxxabi/src/CMakeLists.txt:79]: I'd leave a `TODO` comment to use https://cmake.org/cmake/help/latest/variable/CMAKE_LINK_LIBRARY_USING_FEATURE.html once LLVM requires CMake at least 3.24.

Comment (ilovepi) [reply libcxxabi/src/CMakeLists.txt:81]: done.

**Comment by ilovepi** [comment]
ping. Can someone for @llvm/reviewers-libcxxabi take a look at this please?

**Comment by ilovepi** [comment]
ping

**Review by MaskRay — APPROVED** [review]

File: libcxxabi/src/CMakeLists.txt (hunk 1 above)
Comment (MaskRay) [reply libcxxabi/src/CMakeLists.txt:81]: `TODO: $upper_case` is more common than `TODO `

Comment (ilovepi) [reply libcxxabi/src/CMakeLists.txt:81]: Good catch. Thx.

**Pull request merged**